### PR TITLE
 examples:dns_matching: make it work as a DNS sniffer

### PR DESCRIPTION
**Pull request by pbhole**
Also mentioned about this in my previous PR.

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

**Comment by yonghong-song** [comment]
The current example is not self-contained. For example, it is not clear how to set up the environment to have some meaningful output for the test. Could you help fix it? A complete example will be much more useful. You can either use pyroute2 python library or you can use ip utilities.

File: examples/networking/dns_matching/dns_matching.py [hunk 1]
```diff
@@ -46,12 +52,34 @@ def encode_dns(name):
...
+  UDP_HLEN = 8
+
+  #IP HEADER
+  #calculate ip header length
+  ip_header_length = packet_bytearray[ETH_HLEN]               #load Byte
+  ip_header_length = ip_header_length & 0x0F                  #mask bits 0..3
+  ip_header_length = ip_header_length << 2                    #shift to obtain length
+
+  #calculate payload offset
+  payload_offset = ETH_HLEN + ip_header_length + UDP_HLEN
```
Comment (yonghong-song) [inline examples/networking/dns_matching/dns_matching.py]: How do you know it must be UDP packet?

Comment (pbhole) [reply examples/networking/dns_matching/dns_matching.py]: Because bpf program is filtering-in only UDP packets (only those containing DNS question).

**Comment by pbhole** [comment]
@yonghong-song Thank you for reviewing. I will make it a complete example. However I did not get the context of your suggestion regarding use of pyroute2/ip utilites. Can you please elaborate?

File: examples/networking/dns_matching/dns_matching.py (hunk 1 above)
Comment (yonghong-song) [reply examples/networking/dns_matching/dns_matching.py]: Oh, right. the bpf program drops everything except dns packets.

**Comment by yonghong-song** [comment]
For pyroute2/ip utilities, I mean that you can use them to build a complete test case. But you are free to use whatever utilities to build your example. Thanks!

**Comment by pbhole** [comment]
PR updated. Please review.

**Pull request merged**